crates/services/block_aggregator_api/src/api/protobuf_adapter.rs

```diff
@@ -153,15 +154,26 @@ impl ProtobufAPI {
             tokio::sync::mpsc::channel::<BlockAggregatorQuery<BlockRangeResponse>>(100);
         let server = Server::new(query_sender);
         let addr = url.parse().unwrap();
+        let (shutdown_sender, shutdown_receiver) = tokio::sync::oneshot::channel::<()>();
         let _server_task_handle = tokio::spawn(async move {
-            tonic::transport::Server::builder()
-                .add_service(block_aggregator_server::BlockAggregatorServer::new(server))
-                .serve(addr)
-                .await
-                .unwrap();
+            let service = tonic::transport::Server::builder()
```

Comment:
I know that `ServiceRunner` here is an overkill, but we have GraphQL that looks the same, and it was fiiiiine=D Maybe for consistency plus some logging we could reuse it here as well. It's just an internal service for your main service